.github/workflows/api-docs.yml

```
@@ -0,0 +1,43 @@
+name: Docs
+
+# build the documentation whenever there are new commits on master
+on:
+  push:
+    branches:
+      - master
```

```suggestion
on:
  push:
    branches:
      - master
  workflow_dispatch:
```

I think that would make sense here too, in case you want to rebuild the docs for some reason or the workflow failed.

---

Good idea 👌